lib/core/src/display/navigation/camera_manager.rs

```diff
@@ -0,0 +1,197 @@
+use crate::prelude::*;
+
+use super::mouse_manager::State;
+use super::mouse_manager::MouseManager;
+use super::mouse_manager::MouseClickEvent;
+use super::mouse_manager::MouseWheelEvent;
+use super::mouse_manager::MousePositionEvent;
+use super::mouse_manager::MouseButton;
+use crate::system::web::Result;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector2;
+use std::rc::Rc;
+use std::cell::RefCell;
+use nalgebra::zero;
+
+// ====================
+// === ZoomingEvent ===
+// ====================
+
+/// A struct holding zooming event information.
+pub struct ZoomingEvent {
+    pub focus  : Vector2<f32>,
+    pub amount : f32
+}
+
+impl ZoomingEvent {
+    fn from_mouse_wheel(event:MouseWheelEvent, data:&Rc<RefCell<CameraManagerData>>) -> Self {
+        let amount = event.movement_y;
+        let focus  = data.borrow().mouse_position;
+        Self { amount, focus }
+    }
```

Comment:
I don't like keeping this as a function inside `ZoomEvent`. What happens if someone has a very sensitive mouse and wants to have the `amount` scaled by some number? Translating mouse events to ZoomEvent is definitely responsibility of the ZoomEvent.